erc/ERCS/erc-7562.md

```diff
@@ -180,42 +272,42 @@ To help make sense of these params, note that a malicious paymaster can at most
 * Allowed access to the `EntryPoint` address:
     * **[OP-051]** May call `EXTCODESIZE ISZERO`\
-      This pattern is used to check destination has code before `depositTo` function is called.
+      This pattern is used to check destination has a code before the `depositTo` function is called.
     * **[OP-052]** May call `depositTo(sender)` with any value from either the `sender` or `factory`
     * **[OP-053]** May call the fallback function from the `sender` with any value
     * **[OP-054]** Any other access to the `EntryPoint` is forbidden
 * `*CALL` opcodes:
     * **[OP-061]** `CALL` with `value` is forbidden. The only exception is a call to the `EntryPoint` described above.
     * **[OP-062]** Precompiles:
         * Only allowed the core 9 precompiles.\
-          Specifically the computation precompiles that do not access anything in the blockchain state or environment.
-* **[OP-070] Transient Storage slots defined in [EIP-1153](./eip-1153.md) and accessed using `TLOAD` (`0x5c`) and `TSTORE` (`0x5d`) opcodes
-    are treated exactly like persistent storage (SLOAD/SSTORE)
+          Specifically, the computation precompiles that do not access anything in the blockchain state or environment.
+* **[OP-070]** Transient Storage slots defined in [EIP-1153](./eip-1153) and accessed using `TLOAD` (`0x5c`) and `TSTORE` (`0x5d`) opcodes
+  are treated exactly like persistent storage (SLOAD/SSTORE)
 
 ### Code Rules
 
 * **[COD-010]** Between the first and the second validations, the `EXTCODEHASH` value of any visited address,
-entity or referenced library, may not be changed.\
-If the code is modified, the UserOperation is considered invalid.
+  entity, or referenced library, may not be changed.\
+  If the code is modified, the UserOperation is considered invalid.
 
 ### Storage rules.
 
-The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase are limited as follows:
+The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase is limited as follows:
 
 * **[STO-010]** Access to the "account" storage is always allowed.
 * Access to associated storage of the account in an external (non-entity contract) is allowed if either:
     * **[STO-021]**  The account already exists
     * **[STO-022]**  There is an `initCode` and the `factory` contract is staked.
 * If the entity (`paymaster`, `factory`) is staked, then it is also allowed:
     * **[STO-031]** Access the entity's own storage.
-    * **[STO-032]** Read/Write Access to storage slots that is associated with the entity, in any non-entity contract.
+    * **[STO-032]** Read/Write Access to storage slots that are associated with the entity, in any non-entity contract.
     * **[STO-033]** Read-only access to any storage in non-entity contract.
 * **[STO-040]** `UserOperation` may not use an entity address (`factory`/`paymaster`/`aggregator`) that is used as an "account" in another `UserOperation` in the mempool. \
```

Comment:
right.. this is not a "real" storage rule, but local one - but it is not reputation. we don't have category for that.

Reply:
That's why I didn't think the separation should be reputation rules vs. non-reputation rules, but "objective" vs. "subjective", or "network-wide" vs. "local".  That's the separation that bundler devs should care about, since they can't deviate from network-wide rules but have some discretion on local rules.

And regardless of category, does it need to apply for "in the mempool" and not just "in the same bundle"?  We implemented it at the mempool level in our bundler because it was easier, but from security perspective do we care if it happens but not in the same bundle?